### ISSUE TITLE
remove local golang dependency for ocb binary

### ISSUE THREAD

**Issue by jackgopack4** [open-telemetry/opentelemetry-collector#11382]
**Is your feature request related to a problem? Please describe.**
`ocb` binaries, even as they are compiled and built for specific architectures, still require a golang environment to run. This seems counterintuitive; I might expect `go install` command or compiling from scratch to require golang compiler, but I wouldn't expect a compiled binary to still require golang in the environment

**Describe the solution you'd like**
I would like to remove the golang dependency for the ocb binary, allowing users to run ocb in a minimal (e.g. scratch, alpine, etc.) environment

**Describe alternatives you've considered**
Alternative is adding some explicit mention of golang requirement for `ocb` in documentation, both on github and opentelemetry.io

**Additional context**
I have been working on and submitted https://github.com/open-telemetry/opentelemetry-collector-releases/pull/671 which has a "thicker" image than required due the golang dependency for ocb binary. Making this change would allow a smaller image, such as from `scratch` that the collector distros run on.

**Comment by jackgopack4** [comment]
In my opinion, the quickest solution would be embedding the appropriate version of the golang compiler into `ocb` binary; this would necessarily raise the size of the `ocb` binary by around ~75MB. After looking more into how `ocb` works, anything else would likely require a significant overhaul of `ocb`.

**Comment by Annosha** [comment]
Hi @jackgopack4 I'm an [Outreachy](https://www.outreachy.org/apply/project-selection/#cncf-tracing) applicant. If the documentation part of this issue is open and beginner friendly can I work on it?

**Comment by jackgopack4** [comment]
Absolutely, feel free to take a crack at it. I think it will be up for discussion whether this is a feature the maintainers want to include (certain security concerns with a program that downloads another binary and executes it on your behalf) but if it is a desired feature, it shouldn't be too complicated, code-wise.

**Comment by andrzej-stencel** [comment]
I don't think embedding the Go compiler in the OTel Collector Builder binary is the way to go. The builder's purpose is to generate the sources for a distro and then _use_ the Go compiler to compile the sources. Go compiler is an external dependency of the Builder and in my opinion it should stay like that.

I have commented on https://github.com/open-telemetry/opentelemetry-collector-releases/pull/671 too. Thanks Jack and sorry for the confusion I caused!
